Tasks/GoToolV0/gotool.ts

```diff
@@ -4,135 +4,371 @@ import * as os from 'os';
+        if (parsed.hasMajorMinorPatchRevision()) {
+            throw new Error("Official Go version must be 'major.minor' (e.g. 1.22) or 'major.minor.patch' (e.g. 1.22.3).");
+        }
+        
+        if (parsed.hasMajorMinorPatchOnly()) {
+            // Full patch version specified - use as-is
+            return { filenameVersion: v, cacheVersion: v, toolName: 'go' };
+        }
+        
+        if (parsed.hasMajorMinorOnly()) {
+            // Resolve to latest patch
+            try {
+                const resolved = await getLatestPatchFromGoDev(parsed);
+                return { filenameVersion: resolved, cacheVersion: resolved, toolName: 'go' };
+            } catch (error) {
+                const errorMessage = error instanceof Error ? error.message : String(error);
+                if (/Could not find a stable patch version/i.test(errorMessage)) {
+                    throw new Error(`Requested Go minor ${v} has no stable patch release yet. Specify a full patch once released (e.g. ${v}.0) or use an existing released minor (e.g. 1.23).`);
+                }
+                throw error;
+            }
+        }
+        
+        // Should never reach here due to validation, but defensive programming
+        throw new Error("Official Go version must be 'major.minor' (e.g. 1.22) or 'major.minor.patch' (e.g. 1.22.3).");
+    }
+
+    if (isMicrosoft) {
+        // Microsoft aka channel: accepts major.minor, major.minor.patch, or major.minor.patch-revision
+        if (parsed.hasMajorMinorOnly()) {
+            // Resolve to latest patch-revision from manifest
+            const resolved = await getMicrosoftLatestFromManifest(parsed);
+            return { filenameVersion: resolved, cacheVersion: resolved, toolName: 'go-aka' };
+        }
+        
+        if (parsed.hasMajorMinorPatchOnly()) {
+            // Patch without revision: resolve to latest revision from manifest
+            const resolved = await getMicrosoftLatestFromManifest(parsed);
+            return { filenameVersion: resolved, cacheVersion: resolved, toolName: 'go-aka' };
+        }
+        
+        if (parsed.hasMajorMinorPatchRevision()) {
+            // Full version with revision specified - use as-is
+            return { filenameVersion: v, cacheVersion: v, toolName: 'go-aka' };
+        }
+        
+        // Should never reach here due to validation, but defensive programming
+        throw new Error("For Microsoft Go builds use 'major.minor' (e.g. 1.25) or 'major.minor.patch' (e.g. 1.25.3 or 1.25.3-1).");
+    }
+
+    throw new Error("Invalid download URL. Only https://go.dev/dl and https://aka.ms/golang/release/latest are allowed.");
+}
+
+/**
+ * Resolves the download URL by checking both task parameter and environment variable.
+ * Task parameter takes precedence over environment variable.
+ * @returns The resolved download URL or undefined if neither is set
+ */
+function resolveDownloadUrl(): string | undefined {
```

Comment:
optional: It might be my personal preference, but it's always easier to read code in natural way (from up to down). In other words, I would put definition of this function above `getGo` function, because it's called earlier. With that approach you will avoid jumping up and down when reading/debugging code.

```
const downloadUrl = resolveDownloadUrl();
const resolvedVersion = await getGo(version, downloadUrl);
```